test/e2e/helpers.go

```diff
@@ -0,0 +1,116 @@
+// +build integration interactive
+
+package e2e
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/efficientgo/e2e"
+	"github.com/efficientgo/tools/core/pkg/testutil"
+	"github.com/observatorium/api/test/testtls"
+)
+
+// Generates certificates and copies static configuration to the shared directory.
+func prepareConfigsAndCerts(t *testing.T, testType testType, e e2e.Environment) {
+	testutil.Ok(
+		t,
+		testtls.GenerateCerts(
+			filepath.Join(e.SharedDir(), certsSharedDir),
+			getContainerName(t, testType, "observatorium_api"),
+			[]string{getContainerName(t, testType, "observatorium_api"), "127.0.0.1"},
+			getContainerName(t, testType, "dex"),
+			[]string{getContainerName(t, testType, "dex"), "127.0.0.1"},
+		),
+	)
+
+	testutil.Ok(t, exec.Command("cp", "-r", "../config", filepath.Join(e.SharedDir(), configSharedDir)).Run())
+}
+
+// obtainToken obtains a bearer token needed for communication with the API.
+func obtainToken(endpoint string, tlsConf *tls.Config) (string, error) {
+	type token struct {
+		IDToken string `json:"id_token"`
+	}
+
+	data := url.Values{}
+	data.Add("grant_type", "password")
+	data.Add("username", "admin@example.com")
+	data.Add("password", "password")
+	data.Add("client_id", "test")
+	data.Add("client_secret", "ZXhhbXBsZS1hcHAtc2VjcmV0")
+	data.Add("scope", "openid email")
+
+	r, err := http.NewRequest(http.MethodPost, "https://"+endpoint+"/dex/token", strings.NewReader(data.Encode()))
+	if err != nil {
+		return "", fmt.Errorf("cannot create new request: %v\n", err)
+	}
+	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	c := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConf,
+		},
+	}
+
+	res, err := c.Do(r)
+	if err != nil {
+		return "", fmt.Errorf("request failed: %v\n", err)
+	}
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return "", fmt.Errorf("cannot read body: %v\n", err)
+	}
+
+	var t token
+	if err := json.Unmarshal(body, &t); err != nil {
+		return "", fmt.Errorf("cannot unmarshal token : %v\n", err)
+	}
+
+	return t.IDToken, nil
+}
+
+func getContainerName(t *testing.T, testType testType, serviceName string) string {
+	switch testType {
+	case logs:
+		return envLogsName + "-" + serviceName
+	case metrics:
+		return envMetricsName + "-" + serviceName
+	case interactive:
+		return envInteractive + "-" + serviceName
+	default:
+		t.Fatal("invalid test type provided")
+		return ""
+	}
+}
+
+func getTLSClientConfig(t *testing.T, e e2e.Environment) *tls.Config {
+	cert, err := ioutil.ReadFile(filepath.Join(filepath.Join(e.SharedDir(), certsSharedDir, "ca.pem")))
```

Comment:
```suggestion
	cert, err := ioutil.ReadFile(filepath.Join(e.SharedDir(), certsSharedDir, "ca.pem"))
```

Looks like one filepath.Join is redundant.

Reply:
:facepalm: